From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim van Gompel <timvangompel@gmail.com>
Date: Mon, 13 Dec 2021 10:43:16 +0100
Subject: [PATCH] fix


diff --git a/src/main/java/net/minecraft/world/entity/projectile/ThrowableProjectile.java b/src/main/java/net/minecraft/world/entity/projectile/ThrowableProjectile.java
index c0e04b2a8fef40920c151f54180e6f3b62e66e1f..a2e67e2e031526a5858777c16caeb12546c42450 100644
--- a/src/main/java/net/minecraft/world/entity/projectile/ThrowableProjectile.java
+++ b/src/main/java/net/minecraft/world/entity/projectile/ThrowableProjectile.java
@@ -90,7 +90,7 @@ public abstract class ThrowableProjectile extends Projectile {
         float f;
 
         if (this.isInWater()) {
-            this.remove(this.removalReason.KILLED);
+            this.discard();
             f = 0F;
         } else {
             f = 0.99F;
